vllm/entrypoints/openai/cli_args.py

```diff
@@ -82,220 +83,269 @@ def __call__(
+            default=None,
+            nargs='+',
+            action=LoRAParserAction,
+            help=frontend_kwargs["lora_modules"]["help"])
+
+        # Special case: Prompt adapters need custom parser action
+        frontend_group.add_argument(
+            "--prompt-adapters",
+            type=optional_type(str),
+            default=None,
+            nargs='+',
+            action=PromptAdapterParserAction,
+            help=frontend_kwargs["prompt_adapters"]["help"])
+
+        frontend_group.add_argument("--chat-template",
+                                    **frontend_kwargs["chat_template"])
+        frontend_group.add_argument(
+            "--chat-template-content-format",
+            **frontend_kwargs["chat_template_content_format"])
+
+        frontend_group.add_argument("--response-role",
+                                    **frontend_kwargs["response_role"])
+        frontend_group.add_argument("--ssl-keyfile",
+                                    **frontend_kwargs["ssl_keyfile"])
+        frontend_group.add_argument("--ssl-certfile",
+                                    **frontend_kwargs["ssl_certfile"])
+        frontend_group.add_argument("--ssl-ca-certs",
+                                    **frontend_kwargs["ssl_ca_certs"])
+        frontend_group.add_argument("--enable-ssl-refresh",
+                                    **frontend_kwargs["enable_ssl_refresh"])
+        frontend_group.add_argument("--ssl-cert-reqs",
+                                    **frontend_kwargs["ssl_cert_reqs"])
+        frontend_group.add_argument("--root-path",
+                                    **frontend_kwargs["root_path"])
+
+        # Special case: Middleware needs append action
+        frontend_group.add_argument("--middleware",
+                                    type=optional_type(str),
+                                    action="append",
+                                    default=[],
+                                    help=frontend_kwargs["middleware"]["help"])
+
+        frontend_group.add_argument(
+            "--return-tokens-as-token-ids",
+            **frontend_kwargs["return_tokens_as_token_ids"])
+        frontend_group.add_argument(
+            "--disable-frontend-multiprocessing",
+            **frontend_kwargs["disable_frontend_multiprocessing"])
+        frontend_group.add_argument(
+            "--enable-request-id-headers",
+            **frontend_kwargs["enable_request_id_headers"])
+        frontend_group.add_argument(
+            "--enable-auto-tool-choice",
+            **frontend_kwargs["enable_auto_tool_choice"])
+
+        # Special case: Tool call parser needs custom metavar
+        valid_tool_parsers = ToolParserManager.tool_parsers.keys()
+        frontend_group.add_argument(
+            "--tool-call-parser",
+            type=str,
+            metavar="{" + ",".join(valid_tool_parsers) +
+            "} or name registered in "
+            "--tool-parser-plugin",
```

Comment:
Could this not be done via choices?

Reply:
Ok I agree that we cannot use choices if we need to be able to pass an arbitrary value from `--tool-parser-plugin`